apps/designer/app/canvas/canvas.tsx

```diff
@@ -117,7 +117,7 @@ export const Canvas = ({ data }: CanvasProps): JSX.Element | null => {
   }
   const isDesignerReady = useSubscribeDesignerReady();
   useInitializeBreakpoints(data.breakpoints);
-  globalStyles();
+  useGlobalStyles({ assets: data.assets })();
```

Comment:
Why this strange API where hook returns a function that we call on every render?
Also, calling functions in render is against React conventions unless it's a hook or a pure function.

Reply:
Yeah, there are basically 2 functions

globalCss() that returns a function, where globalCss is supposed to be called in the module scope only once and the function it returns would be called inside the component.

Since I need to call globalCss when component renders first time, because font-faces are based on data, I needed a way to memorize that call, so I wrapped it into useCallback. useEffect doesn't work here well, since we need that for ssr.

I could use useMemo (with side effect, which would be wrong too) or I could call the useCallback directly inside useGlobalStyles hook so that that second invocation is hidden in the hook.

Any other options?

Reply:
Actually I just noticed its being called many times even with the same assets object, so its not memorizing properly, I am looking for a better solution now

Reply:
Found a better way using useRef, still hiding the problem with the underlying  globalCss API, but at least this is not leaking outside of the hook

Reply:
another option is something like this useEffect + useInsertionEffect

https://github.com/cssinjs/jss/blob/master/packages/react-jss/src/createUseStyles.js#L15